infrastructures/infrastructure-azure/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AzureInfrastructure.java

```diff
@@ -301,29 +295,66 @@ public void acquireNode() {
-                                                                    privateNetworkCIDR,
-                                                                    staticPublicIP);
-
-        LOGGER.info("Instances ids created : " + instancesIds);
+        boolean instanceIdSaved = false;
+
+        if (compareAndSetAlreadyCreatedFlag(false, true)) {
+            // it is a fresh deployment: create or retrieve all instances
+            instancesIds = connectorIaasController.createAzureInstances(getInfrastructureId(),
+                                                                        instanceTag,
+                                                                        image,
+                                                                        numberOfInstances,
+                                                                        vmUsername,
+                                                                        vmPassword,
+                                                                        vmPublicKey,
+                                                                        vmSizeType,
+                                                                        resourceGroup,
+                                                                        region,
+                                                                        privateNetworkCIDR,
+                                                                        staticPublicIP);
+            LOGGER.info("Instances ids created or retrieved : " + instancesIds);
+        } else {
+            // if the infrastructure was already created, then wee need to
+            // look at the free instances, if any (the ones on which no node
+            // run. In the current implementation, this can only happen when
+            // nodes are down. Indeed if they are all removed on purpose, the
+            // instance should be shut down). Note that in this case, if the
+            // free instances map is empty, no script will be run at all.
+            Map<String, Integer> freeInstancesMap = getFreeInstancesMapCopy();
+            instancesIds = freeInstancesMap.keySet();
+            LOGGER.info("Instances ids previously saved which require script re-execution: " + instancesIds);
+            instanceIdSaved = true;
+        }
 
+        // execute script on instances to deploy or redeploy nodes on them
         for (String instanceId : instancesIds) {
-
             String fullScript = generateScriptFromInstanceId(instanceId);
-            connectorIaasController.executeScript(getInfrastructureId(), instanceId, Lists.newArrayList(fullScript));
+            try {
+                connectorIaasController.executeScript(getInfrastructureId(),
+                                                      instanceId,
+                                                      Lists.newArrayList(fullScript));
+            } catch (Exception e) {
```

Comment:
Why not catching the new `ScriptNotExecutedException`?

Reply:
yes